test/jdk/javax/net/ssl/TLSCommon/TLSWithEdDSA.java

```diff
@@ -0,0 +1,734 @@
+                socket.setEnabledCipherSuites(pVal.split(":"));
+            }
+
+            pVal = serverParameters.get(ParamType.CLIAUTH);
+            socket.setWantClientAuth("WANT".equalsIgnoreCase(pVal));
+            socket.setNeedClientAuth("NEED".equalsIgnoreCase(pVal));
+        } catch (SocketException se) {
+            throw new RuntimeException(se);
+        }
+    }
+
+
+    @Override
+    protected void runServerApplication(SSLSocket socket) throws Exception {
+        InputStream sslIS = socket.getInputStream();
+        OutputStream sslOS = socket.getOutputStream();
+
+        sslIS.read();
+        sslOS.write(85);
+        sslOS.flush();
+
+        if (serverChecker != null) {
+            serverChecker.check(socket);
+        }
+    }
+
+    @Override
+    protected void runClientApplication(SSLSocket socket) throws Exception {
+        InputStream sslIS = socket.getInputStream();
+        OutputStream sslOS = socket.getOutputStream();
+
+        sslOS.write(280);
+        sslOS.flush();
+        sslIS.read();
+
+        if (clientChecker != null) {
+            clientChecker.check(socket);
+        }
+    }
+
+    public static void main(String[] args) throws Exception {
+        if (DEBUG) {
+            System.setProperty("javax.net.debug", "ssl:handshake");
+        }
+        certFac = CertificateFactory.getInstance("X.509");
+        String testFormat;
+
+        System.out.println("===== Test KeyManager alias retrieval =====");
+        keyManagerTests(DEF_ALL_EE, "EdDSA",
+                new String[] {"ee_ed25519", "ee_ed448"});
+
+        testFormat =
+                "===== Basic Ed25519 Server-side Authentication: %s =====\n";
+        serverParameters.put(ParamType.KSENTRIES, "EE_ED25519:EE_RSA_2048");
+        runtest(testFormat, isPeerEd25519, null, null, null);
+
+        testFormat =
+                "===== Basic Ed448 Server-side Authentication: %s =====\n";
+        serverParameters.put(ParamType.KSENTRIES, "EE_ED448:EE_RSA_2048");
+        runtest(testFormat, isPeerEd448, null, null, null);
+
+        testFormat = "===== EC favored over EdDSA by default: %s =====\n";
+        serverParameters.put(ParamType.KSENTRIES,
+                "EE_ED25519:EE_ECDSA_SECP521R1");
+        runtest(testFormat, isPeerP521, null, null, null);
+
+        testFormat = "===== Override EC favoring by alias: %s =====\n";
+        serverParameters.put(ParamType.CERTALIAS, "EE_ED25519");
+        runtest(testFormat, isPeerEd25519, null, null, null);
+        serverParameters.remove(ParamType.CERTALIAS);
+
+        testFormat = "===== EdDSA Client Authentication: %s =====\n";
+        serverParameters.put(ParamType.KSENTRIES, "EE_RSA_2048");
+        serverParameters.put(ParamType.CLIAUTH, "NEED");
+        clientParameters.put(ParamType.KSENTRIES, "EE_ED25519");
+        runtest(testFormat, null, null, isPeerEd25519, null);
+    }
+
+    private static void keyManagerTests(String keyStoreSpec, String keyType,
```

Comment:
Java method name is normally an action. What do you think if update to testKeyManager()?

Reply:
Sure, easy enough to do.